tests/components/anthemav/test_config_flow.py

```diff
@@ -0,0 +1,206 @@
+
+
+async def test_form_invalid_model(
+    hass: HomeAssistant, mock_connection_create: AsyncMock, mock_anthemav: AsyncMock
+) -> None:
+    """Test we handle when receiving and invalid AVR model."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
+
+    mock_anthemav.protocol.model = UNKNOWN_MODEL
+
+    with patch(
+        "homeassistant.components.anthemav.async_setup_entry",
+        return_value=True,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+                "port": 14999,
+                "name": "Anthem Av",
+            },
+        )
+
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "cannot_receive_deviceinfo"}
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "anthemav.Connection.create",
+        side_effect=OSError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+                "port": 14999,
+                "name": "Anthem Av",
+            },
+        )
+
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_unknown_error(hass: HomeAssistant) -> None:
+    """Test we handle any errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "anthemav.Connection.create",
+        side_effect=Exception,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+                "port": 14999,
+                "name": "Anthem Av",
+            },
+        )
+
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_import_configuration(
+    hass: HomeAssistant, mock_connection_create: AsyncMock, mock_anthemav: AsyncMock
+) -> None:
+    """Test we import existing configuration."""
+    config = {
+        "host": "1.1.1.1",
+        "port": 14999,
+        "name": "Anthem Av",
+    }
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=config
+    )
+
+    await hass.async_block_till_done()
```

Comment:
I think this may not be needed, but I am not positive.  Unless we may also want to assert async_setup_entry is called on import too?

Reply:
You are correct, we don't need to block here.
I'm not mocking async_setup_entry, not sure how/why to assert if called. Or do you suggest to mock async_setup_entry instead and just check that it is called? I would rather test the full logic on that one.

Reply:
OK let's just remove the unnecessary block then and leave the rest as is.